In [1]:
from sklearn.linear_model import LinearRegression

clf = LinearRegression()
clf.fit([[0, 0], [1, 1], [2, 2]], [0, 1, 2])  # 模型训练
'''
y = 0.5*x1 + 0.5*x2
'''
pre = clf.predict([[3, 3]])   # 模型预测

In [2]:
clf.coef_

array([0.5, 0.5])

In [3]:
clf.intercept_

1.1102230246251565e-16

In [4]:
print(pre)

[3.]


In [5]:
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report
data = pd.read_csv('LogisticRegression.csv')

In [6]:
data.head(5)

admit  gre   gpa  rank
0      0  380  3.61     3
1      1  660  3.67     3
2      1  800  4.00     1
3      1  640  3.19     4
4      0  520  2.93     4

In [7]:
data.shape

(400, 4)

In [8]:
data_tr, data_te, label_tr, label_te = train_test_split(data.iloc[:, 1:], data['admit'], test_size=0.2)
clf = LogisticRegression()
clf.fit(data_tr, label_tr)
pre = clf.predict(data_te)
res = classification_report(label_te, pre)
print(res)

              precision    recall  f1-score   support

           0       0.71      0.93      0.81        54
           1       0.60      0.23      0.33        26

    accuracy                           0.70        80
   macro avg       0.66      0.58      0.57        80
weighted avg       0.68      0.70      0.65        80



### 1. 类别分析（Class 0 和 Class 1）
### Class 0（类别 0）
### Precision（精确率） = 0.71
### 含义：模型预测为类别 0 的样本中，71% 是真实的类别 0。
### 问题：存在一定误判（29% 的预测错误）。

### Recall（召回率） = 0.93
### 含义：真实为类别 0 的样本中，93% 被模型正确识别。
### 优点：模型对类别 0 的覆盖能力很强。

### F1-score（调和均值） = 0.81
### 含义：Precision 和 Recall 的平衡值（越接近 1 越好）。
### 结论：类别 0 的整体表现优秀。